### PR TITLE
Make default locale a compile time option

### LINKED ISSUE #11259: What are the "clear indications" of C.UTF-8 existing soon upstream?
v240 NEWS mentions that systemd now depends on C.UTF-8 "clear indications that upstream
glibc is going to make it available too." I'd like to understand what those indications are, because it's entirely non-obvious to me.

The last time I see this discussed on [libc-alpha](https://sourceware.org/ml/libc-alpha/2015-02/msg00293.html) is 2015. Similarly, there's a proposal on the [glibc wiki](https://sourceware.org/glibc/wiki/Proposals/C.UTF-8) which was added in 2015, but there's been no updates/followup since then.

The wiki page for the glibc 2.29 [release](https://sourceware.org/glibc/wiki/Release/2.29) makes no mention of this new locale, but does mention that glibc is about a month away (Feb 1 2019) from a release. I'd expect that given multiple distros have invented their own C.UTF-8 and that the indications of its upstream inclusion are now clear, that this would be mention-worthy. Cloning the glibc git repo yields no evidence that such a locale has been added.

Indeed, what have I missed in the intervening 3 years? IMO, the news item should have linked to the obvious indicator that this will be coming in a future release of glibc. Otherwise, this just looks like a losing bet to me.

### PR DESCRIPTION
**Pull request by falconindy**
Default to a locale that's guaranteed to exist everywhere, but let
distros override this with something more exotic if they choose to.

Closes #11259.

### CODE REVIEW

**Comment by yuwata** [comment]
LGTM. Please update several tests.

**Comment by yuwata** [comment]
If it is possible, then I prefer that meson checks that `C.UTF-8` exists or not, and use `C.UTF-8` if it exists.

File: meson_options.txt [hunk 1]
```diff
@@ -195,6 +195,8 @@ option('default-kill-user-processes', type : 'boolean',
        description : 'the default value for KillUserProcesses= setting')
 option('gshadow', type : 'boolean',
        description : 'support for shadow group')
+option('default-locale', type : 'string', value : 'C',
```
Comment (poettering) [inline meson_options.txt:198]: please default to C.UTF-8 here.

**Comment by yuwata** [comment]
@falconindy I've added a follow-up commit in #11291. PTAL.

File: meson_options.txt (hunk 1 above)
Comment (falconindy) [reply meson_options.txt:198]: I have no interest in defaulting to a locale that does not exist. Anyways, yuwata's patch provides a better solution here.

**Comment by falconindy** [comment]
> LGTM. Please update several tests, otherwise CIs cannot pass.

Not clear what tests need to be updated. Looking at the amd64 autopkgtest results, I see:

```
OK:       474
FAIL:       0
SKIP:      13
TIMEOUT:    0
```

File: meson_options.txt (hunk 1 above)
Comment (Scimmia22) [reply meson_options.txt:198]: Sane defaults, by the very definition of the phrase, are things that work everywhere.

**Pull request closed without merging**